RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -111,26 +118,140 @@ std::pair<std::vector<T>, std::vector<T>> categorise(
 
   return output;
 }
+template <typename T>
+std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> L1TJetsMatching<T>::categoriseVBFPlus2CentralJets(
+    const std::vector<T>& pfMatchedJets, double pt1, double pt2, double pt3, double Mjj) const {  //60, 30, 50, 500
+  std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> output;
+  unsigned int i1 = 0;
+  unsigned int i2 = 0;
+
+  double mjj = 0;
+  if (pfMatchedJets.size() > 1) {
+    for (unsigned int i = 0; i < pfMatchedJets.size() - 1; i++) {
+      const T& myJet1 = (pfMatchedJets)[i];
+
+      for (unsigned int j = i + 1; j < pfMatchedJets.size(); j++) {
+        const T& myJet2 = (pfMatchedJets)[j];
+
+        const double mjj_test = (myJet1.p4() + myJet2.p4()).M();
+
+        if (mjj_test > mjj) {
+          mjj = mjj_test;
+          i1 = i;
+          i2 = j;
+        }
+      }
+    }
+
+    const T& myJet1 = (pfMatchedJets)[i1];
+    const T& myJet2 = (pfMatchedJets)[i2];
+
+    std::vector<T> vec4jets;
+    vec4jets.reserve(4);
+    std::vector<T> vec5jets;
+    vec5jets.reserve(5);
+    std::vector<T> vec6jets;
+    vec6jets.reserve(6);
+    if (pfMatchedJets.size() > 3) {
+      if ((mjj > Mjj) && (myJet1.pt() >= pt3) && (myJet2.pt() > pt2)) {
+        vec4jets.push_back(myJet1);
+        vec4jets.push_back(myJet2);
+        unsigned int i3 = 0;
+        unsigned int i4 = 0;
+        bool filledi3 = false;
+        for (unsigned int i = 0; i < pfMatchedJets.size(); i++) {
+          if (i == i1)
+            continue;
+          if (i == i2)
+            continue;
+          if (!filledi3) {
+            i3 = i;
+            filledi3 = true;
+          } else {
+            i4 = i;
+            break;
+          }
+        }
+        const T& myJet3 = pfMatchedJets[i3];
+        const T& myJet4 = pfMatchedJets[i4];
+        vec4jets.push_back(myJet3);
+        vec4jets.push_back(myJet4);
+      }
+
+      if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() < pt3) && (myJet1.pt() > pt2) &&
+          (myJet2.pt() > pt2)) {  //60, 30, 50, 500
+
+        std::vector<unsigned int> idx_jets;
+        idx_jets.clear();
+
+        for (unsigned int i = 1; i < pfMatchedJets.size(); i++) {  //skip highest pT jet
+          if (i == i1 || i == i2)
+            continue;
+          if ((pfMatchedJets)[i].pt() > pt2) {
+            idx_jets.push_back(i);
+          }
+        }
+        if (idx_jets.size() == 2) {
+          const T& thirdjet = pfMatchedJets[0];
+          const T& fourthjet = pfMatchedJets[idx_jets[0]];
+          const T& fifthjet = pfMatchedJets[idx_jets[1]];
+
+          vec5jets.push_back(myJet1);
+          vec5jets.push_back(myJet2);
+          vec5jets.push_back(thirdjet);
```

Comment:
Can it happen that `i1 == 0`, and then `myJet1` and `thirdjet` are actually the same jet ?

Reply:
This should not be possible with the requirement on l.181 (`myJet1.pt() < pt1`), as there should be at least one jet with `pT>pT1` for events reaching this filter

Reply:
> there should be at least one jet with pT>pT1 for events reaching this filter

Okay, but the filter itself does not know that, and it doesn't enforce this requirement, right ?

Reply:
Right, I adapted the logic, not excluding `idx_jets==0` anymore. This way it should not be possible to use the same jet twice when filling the vectors